### DISCUSSION TITLE
[choco] Release not pushed

### DISCUSSION THREAD

**Discussion by mthmulders** [jreleaser/jreleaser#847] [Q&A] [answered]
I finally found some time to update my project, [MCS](https://github.com/mthmulders/mcs), to JReleaser 1.0.0. Long overdue, I know.... I was surprised with two things, I'll open separate discussions for them. This is 2/2.

After upgrading from JReleaser 0.1.0 to 1.0.0, and pushing a new MCS release, I noticed that the Chocolatey change had not been pushed, despite the log saying

>  [INFO]     [chocolatey] publishing mcs distribution
[INFO]     [chocolatey] setting up repository mthmulders/chocolatey-bucket
[INFO]     [chocolatey] pushing to mthmulders/chocolatey-bucket

Should we consider this an unfortunate chain of events in the universe, or maybe is this a sign that tells us something's broken? ;-)

**Comment by aalmiray** [comment]
Is the repo https://github.com/mthmulders/chocolatey-bucket missing something for `0.2.1`? One thing I noticed is that there's no tag.

**Reply by mthmulders** [reply]
The tag is probably missing because I didn't create it. Since the commit wasn't pushed from https://github.com/mthmulders/mcs/actions/runs/2530990562, I [manually created](https://github.com/mthmulders/chocolatey-bucket/commit/99f0f786a63bd42b19aaf4e261aca668917470fb) it, but I didn't create a tag. Will add that, just for the record.

But the thing is, JReleaser *says* it pushes to the chocolatey-bucket, but the commit never got there.

**Comment by aalmiray** [comment]
I forked your repository and changed the chocolatey configuration to point to `aalmiray/chocolatey-mcs` then pushed a full release.
The Chocolatey package was successfully pushed to the repository as witnessed by https://github.com/aalmiray/chocolatey-mcs

The push action failed because there's no API key configured yet, this is expected.

**Reply by mthmulders** [reply]
Hmm. I see. I'll keep an eye on it for the next release. Let's say the stars weren't aligned properly when I did the 0.2.1 release :-)

**Reply by mthmulders** [reply]
Stars still not aligned, it seems... for v0.2.2, the commit didn't get pushed either.

Excerpt from JReleasers trace.log:

```
[INFO]      [chocolatey] publishing mcs distribution
[DEBUG]     [chocolatey] creating props for mcs/chocolatey
[DEBUG]     [chocolatey] filling distribution properties into props
[DEBUG]     [chocolatey] filling git properties into props
[DEBUG]     [chocolatey] filling artifact properties into props
[DEBUG]     [chocolatey] filling packager properties into props
[INFO]      [chocolatey] setting up repository mthmulders/chocolatey-bucket
[DEBUG]     [chocolatey] locating repository mthmulders/chocolatey-bucket
[DEBUG]     [chocolatey] lookup repository mthmulders/chocolatey-bucket
[DEBUG]     [chocolatey] lookup repository mthmulders/chocolatey-bucket
[DEBUG]     [chocolatey] cloning https://github.com/mthmulders/chocolatey-bucket.git
[DEBUG]     [chocolatey] copying files from target/jreleaser/package/mcs/chocolatey
[DEBUG]     [chocolatey] copying files from /home/runner/work/mcs/mcs/target/jreleaser/package/mcs/chocolatey to /tmp/jreleaser-chocolatey-bucket941300168901046930
[DEBUG]     [chocolatey] setting up commit
[DEBUG]     [chocolatey] tagging local repository with v0.2.2
[INFO]      [chocolatey] pushing to mthmulders/chocolatey-bucket
[DEBUG]     [chocolatey] pushing commit to remote
```

**Reply by aalmiray** [reply]
Your build file is pulling jreleaser-maven-plugin:1.0.0 but it should be 1.2.0. Did you try using the latest?

**Reply by mthmulders** [reply]
Upgraded to 1.2.0 this evening, and still, the release is not getting pushed:

```
[INFO]  Publishing distributions
[INFO]    - Publishing mcs distribution
...
[INFO]      [chocolatey] publishing mcs distribution
[DEBUG]     [chocolatey] creating props for mcs/chocolatey
[DEBUG]     [chocolatey] filling distribution properties into props
[DEBUG]     [chocolatey] filling git properties into props
[DEBUG]     [chocolatey] filling artifact properties into props
[DEBUG]     [chocolatey] filling packager properties into props
[INFO]      [chocolatey] setting up repository mthmulders/chocolatey-bucket
[DEBUG]     [chocolatey] locating repository mthmulders/chocolatey-bucket
[DEBUG]     [chocolatey] lookup repository mthmulders/chocolatey-bucket
[DEBUG]     [chocolatey] lookup repository mthmulders/chocolatey-bucket
[DEBUG]     [chocolatey] cloning https://github.com/mthmulders/chocolatey-bucket.git
[DEBUG]     [chocolatey] copying files from target/jreleaser/package/mcs/chocolatey
[DEBUG]     [chocolatey] copying files from /home/runner/work/mcs/mcs/target/jreleaser/package/mcs/chocolatey to /tmp/jreleaser-chocolatey-bucket9313869035706870613
[DEBUG]     [chocolatey] setting up commit
[DEBUG]     [chocolatey] tagging local repository with v0.2.3
[INFO]      [chocolatey] pushing to mthmulders/chocolatey-bucket
[DEBUG]     [chocolatey] pushing commit to remote
```

What bugs me here is that, from the JReleaser log, it _seems_ as if everything is fine. There's no mention of any error or failure. But the [commit history of the chocolatey bucket](https://github.com/mthmulders/chocolatey-bucket/commits/master) doesn't have any commit for v0.2.3.

If there's anything else I could do to help troubleshoot, I'd love to help out. But I'm a bit out of ideas at this point.

**Reply by aalmiray** [reply]
Alright. Gave it another try. Updated my for of `mcs` and pushed `v0.3.0` (with bogus binaries)

https://github.com/aalmiray/mcs/releases/tag/v0.3.0

The release also pushed to `aalmiray/chocolatey-mcs` (I tweaked the config to use this name instead

https://github.com/aalmiray/chocolatey-mcs

I think the problem may be the GH token in use by `mcs`. This release was made locally using a token that has lots of permissions granted. I think `repo/public_repo` should be enough however I have ticked all of `repo`. What's strange is that if this were an authentication/permissions issue then the git push should fail.

![pat](https://user-images.githubusercontent.com/13969/196051902-dc3afd49-f563-463a-9786-2579535664e5.jpg)

**Reply by mthmulders** [reply]
Thanks for diving into this, @aalmiray. The token I'm using has the exact same scope (`repo/*`) as yours.

> What's strange is that if this were an authentication/permissions issue then the git push should fail.

True, and that's what worries me a bit. I wouldn't mind if a `git push` failed - but at least tell me it did. Please do not pretend all went fine when it did not.

For a minute, I thought "maybe somehow, JReleaser created an empty commit" - then there would be nothing to push. But the *trace.log* also states it created a tag and that tag is also not present in the GitHub repository.

**Reply by aalmiray** [reply]
Perhaps we can have a look at your configuration during JFall. There's something I'm missing but I can't quite figure out what is it.

**Reply by mthmulders** [reply]
> Perhaps we can have a look at your configuration during JFall.

Sounds like a plan, let's try to find some time then!

**Reply by mthmulders** [reply]
Running `mvn -X` gave some interesting output:

```
[DEBUG] git< unpack ok
[DEBUG] git< ng refs/heads/master refusing to allow a Personal Access Token to create or update workflow `.github/workflows/push.yml` without `workflow` scope
[DEBUG] git< ng refs/tags/v0.2.4 refusing to allow a Personal Access Token to create or update workflow `.github/workflows/push.yml` without `workflow` scope
```

I've updated my access token to include the `workflow` scope and now it works flawlessly!

**Comment by mthmulders** [comment, marked as answer]
It turned out my access token did not include the `workflow` scope. After I have updated the scopes for the token, pushing to my Chocolatey bucket works like a charm.
